src/main/java/com/faforever/client/replay/ReplayService.java

```diff
@@ -513,14 +513,14 @@ private void runFafReplayFile(Path path) throws IOException {
 
     // For some reason in the coop replay the map name is null in the metadata
```

Comment:
seems to work

Reply:
Was absolutely not what I meant XD I meant write a unit test

Reply:
Concerning what u did I don't know if that's a good idea. No idea what pros and cons are too that

Reply:
Maybe @micheljung @Wesmania know what pros and cons are about reading the map from the replay file or the replayInfo stored in the databse.

Reply:
It's the right thing to do for randomized maps, since these will never be stored in the DB. The coop map issue is here: https://github.com/FAForever/server/issues/372
I don't know which is right for all the rest tbh.

Reply:
I tested it and it worked for maps in the vault as well. I think an issue would arise if for some reason the map in the lua header was different than the vault or if the map table structure is intended to be changed. Also I believe the replayinfo that is being read here is stored in the replay file as well just it is the part in the faf header vs the lua header.

Reply:
Hmm well I mean if it works better let's do it this way right.